src/libos/src/vm/process_vm.rs

```diff
@@ -248,7 +248,7 @@ impl<'a, 'b> ProcessVMBuilder<'a, 'b> {
                 );
 
                 // Set the remaining part to zero based on alignment
-                empty_end_offset = align_up(mem_start_offset + file_size, alignment);
+                empty_end_offset = align_up(mem_start_offset + max(file_size, mem_size), alignment);
```

Comment:
file_size should always smaller than mem_size. We can add assertion to confirm it.

Reply:
Done. Thanks for your review.